components/style/media_queries.rs

```diff
@@ -243,19 +243,13 @@ pub fn parse_media_query_list(input: &mut Parser) -> MediaList {
     }
 
     let mut media_queries = vec![];
-    let mut found_invalid = false;
     loop {
         match input.parse_until_before(Delimiter::Comma, MediaQuery::parse) {
-            Ok(mq) => if !found_invalid {
+            Ok(mq) => {
                 media_queries.push(mq);
             },
-            Err(..) => if !found_invalid {
-                media_queries.clear();
+            Err(..) => {
                 media_queries.push(MediaQuery::never_matching());
```

Comment:
Regarding the `@media 2px`, nope, this is exactly Gecko's behavior (and Blink's fwiw).

In the following test case:

```html
<style>
@media 2px {
  #test { color: red; }
}
</style>
<p id="test">Shouldn't be red</p>
<script>
  console.log(document.styleSheets[0].cssRules[0].cssText);
</script>
```

Both engines log:

```
@media not all {
  #test { color: red; }
}
```